src/main/scala/com/atamaniv/Main.scala

```diff
@@ -1,21 +1,40 @@
 package com.atamaniv
 
 import akka.actor.ActorSystem
+import com.atamaniv.Messages.{PrintMessage, StartApplication}
 
-object Main extends Greeting with App {
-  println(startMessage)
+import scala.reflect.io.{File, Path}
+
+/***
+  * Main Class of CrimeStatistics app, accept one argument, the path of folder,
+  * It checks if folder exists and print back an error otherwise.
+  * CrimeSupervisor actor is created here.
+  * */
+
+object Main extends App {
+
+  final val SYSTEM_NAME = "crime-system"
+  final val MAIN_ACTOR_NAME = "crime-supervisor"
+  var isStarted: Boolean = false //For tests purposes only
 
   override def main(args: Array[String]): Unit = {
-    val system = ActorSystem("crime-system")
 
-    try {
-      val supervisor = system.actorOf(IotSupervisor.props(), "crime-supervisor")
-    } finally {
-      println("exit")
+    args.toList match {
+      case head :: Nil =>  if (isDirectoryExists(head)) startApplication(head)
+      else println(s"please provide a correct path to folder, this one $head is not correct")
+      case Nil => println("Please provide absolute path to your csv folder with crime data")
```

Comment:
what happens if 2 and more arguments are passed to your program?

Reply:
> what happens if 2 and more arguments are passed to your program?

Then the head will be taken.

Reply:
Hm, your pattern matching checks for empty list and for list with 1 element. I think for 2 and more elements it will fail with Match error at runtime.

Reply:
Yes, you are right. I have fixed the issue)